libflux/src/core/tests/analyze_test.rs

```diff
@@ -1,11 +1,86 @@
 use core::ast;
-use core::semantic::convert_source;
 use core::semantic::nodes::*;
-use core::semantic::types::{Function, MonoType, SemanticMap, Tvar};
+use core::semantic::types::{Function, MonoType, Property as TypeProperty, Row, SemanticMap, Tvar};
 use core::semantic::walk::{walk_mut, NodeMut};
+use core::semantic::{convert_source, find_var_type};
 
 use pretty_assertions::assert_eq;
 
+#[test]
+fn find_var_ref() {
+    let source = r#"
+vint = v.int + 2
+f = (v) => v.shadow
+g = () => v.sweet
+x = g()
+vstr = v.str + "hello"
+"#;
+    let t = find_var_type(source, "v").expect("Should be able to get a MonoType.");
+    assert_eq!(
+        t,
+        MonoType::Row(Box::new(Row::Extension {
+            head: TypeProperty {
+                k: "int".to_string(),
+                v: MonoType::Int,
+            },
+            tail: MonoType::Row(Box::new(Row::Extension {
+                head: TypeProperty {
+                    k: "sweet".to_string(),
+                    v: MonoType::Var(Tvar(11)),
+                },
+                tail: MonoType::Row(Box::new(Row::Extension {
+                    head: TypeProperty {
+                        k: "str".to_string(),
+                        v: MonoType::String,
+                    },
+                    tail: MonoType::Var(Tvar(22))
+                })),
+            }))
+        }))
+    );
+
+    let serialized = serde_json::to_string_pretty(&t).unwrap();
+    println!("{}", serialized);
```

Comment:
Are these just left over debug statements or did you mean to test the json output?